CommunicationLayer/MessagingFramework/MotorDetailsMessage.cpp

```
@@ -0,0 +1,153 @@
+}
+
+float MotorDetailsMessage::motorCurrentImaginary() const
+{
+    return getFloat(messageData_, MOTOR_CURRENT_IMAGINARY);
+}
+
+float MotorDetailsMessage::backEmfReal() const
+{
+    return getFloat(messageData_, BACK_EMF_REAL);
+}
+
+float MotorDetailsMessage::backEmfImaginary() const
+{
+    return getFloat(messageData_, BACK_EMF_IMAGINARY);
+}
+
+float MotorDetailsMessage::voltageRailSuppply15V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPPLY_15_V);
+}
+
+float MotorDetailsMessage::voltageRailSupply33V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPLY_33_V);
+}
+
+float MotorDetailsMessage::voltageRailSupply19V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPLY_19_V);
+}
+
+float MotorDetailsMessage::heatSinkTemperature() const
+{
+    return getFloat(messageData_, HEAT_SINK_TEMPERATURE);
+}
+
+float MotorDetailsMessage::motorTempterature() const
+{
+    return getFloat(messageData_, MOTOR_TEMPTERATURE);
+}
+
+float MotorDetailsMessage::dspBoardTemperature() const
+{
+    return getFloat(messageData_, DSP_BOARD_TEMPERATURE);
+}
+
+float MotorDetailsMessage::dcBusAmpHours() const
+{
+    return getFloat(messageData_, DC_BUS_AMP_HOURS);
+}
+
+float MotorDetailsMessage::odometer() const
+{
+    return getFloat(messageData_, ODOMETER);
+}
+
+float MotorDetailsMessage::slipSpeed() const
+{
+    return getFloat(messageData_, SLIP_SPEED);
+}
+
+QString MotorDetailsMessage::toString() const
+{
+    QString messageString;
+    if (motorNumber_ == 0x0) {
+        messageString += QString::number(MessageDefines::Motor0Details) + ", ";
+    }
+    else {
+        // For the sake of not halting programs execution, treat all other cases as motor one
+        if (motorNumber != 0x1) {
+            // Log that this was used incorrectly
+            qDebug() << "Invalid motor number, defaulting to motor 1";
+        }
+        messageString += QString::number(MessageDefines::Motor1Details) + ", ";
+    }
```

If the method of distinguishing the two motors is appropriate, is this how you would like to deal with an incorrect motor ID being input? If this is I will of course change it to compare against an enum instead, as this was just a mock up to see what you thought. 
Personally I would prefer to throw an exception in the constructor, as I suspect the only time an incorrect id should be passed into the constructor is when the logic to create the messages is faulty in the first place, and doing something like this might allow that faulty code to make it onto the car if it wasn't caught by my qDebug statement during testing. Let me know what you think is appropriate

---

No I think this is dangerous, The program should not default to a "correct" motor data, it makes a problme easy to ignore in the future. Personally I don't think an Exception is the right way to go either perhaps just follow normal flow of instruction, but don't cover up the mistake by defaulting to motor one.

---

@enochtsang  

In this case there are only two motors defined in MessageDefines (i.e. Motor 0 and Motor 1). So I'm not sure what the normal flow of instruction would be if given an incorrect motor number as it would not be defined. I could implement so it is calculating the package ID via the following
```
messageString += QString::number(MessageDefines::Motor0Details + motorNumber_) + ", ";
``` 

But I personally feel this would be far more difficult to debug. I could do this in conjunction with a debug statement, but on the off chance it didn't cause problems it would suck if faulty code somehow made it's way into the car. 

In any case let me know, I'll go with whatever you decide

---

@enochtsang 

Ok I'm assuming this is the thread you were confused about. so currently my implementation is 
```
if ((motorNumber & 0xe) != 0x0) {
        qDebug() << "Invalid motor number, unexpected behaviour likely";
    }
```
Which, assuming my logic is correct, will output a debug statement if the program receives a motor number other than 0 or 1. I understand now that it should not halt the execution because this is something being sent to Hermes and is not internal to the programs execution, so there is really no way to guarantee this situation doesn't occur. If this is an appropriate way of dealing with the problem let me know (All I'm doing is outputting this debug message before continuing with the normal flow of the program), or if there is something you have in mind that's better I'd be happy to alter this. :)

---

Ok, that's fine. My concern was that if all we had was a debug message, we would wouldn't see the issue in production until we looked at the logs. It'd be better to have something funky happen in the car during live testing if something was coded wrong.